### PR TITLE
Adds multi-target configuration

### PR DESCRIPTION
**Pull request by marrony**
This PR refactor the projects files to use the new `PackageReference` tags instead of use old `packages.config` file. It allows us to create a driver that targets multiple dotnet frameworks. After merged it would require some changes in our Jenkins in order to build/test.

closes https://github.com/fauna/faunadb-csharp/issues/61

### CODE REVIEW

**Comment by marrony** [comment]
With this PR I was able to compile/test the driver in the following dotnet frameworks, both on mono (macOS) and on Windows:

* `netstandard1.5` which is implemented by `netcoreapp1.0` and `netcoreapp1.1`
* `netstandard2.0` which is implemented by `netcoreapp2.0`
* `net45` and `net451`
* `net46` and `net461`
* `net47`

File: FaunaDB.Client/FaunaDB.Client.csproj [hunk 1]
```diff
@@ -1,97 +1,36 @@
...
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
+    <TargetFrameworks>netstandard1.5;netstandard2.0;net45;net451;net46;net461;net47</TargetFrameworks>
+    <PackageVersion>2.0.0-alpha</PackageVersion>
+    <PackageId>FaunaDB.Client</PackageId>
+    <Title>C# Driver for FaunaDB</Title>
+    <Authors>Fauna, Inc.</Authors>
+    <Description>C# Driver for FaunaDB.
+
+See https://fauna.com for more information.</Description>
+    <Copyright>© Fauna, Inc. 2017. Distributed under MPL 2.0 License</Copyright>
```
Comment (ashfire908) [inline FaunaDB.Client/FaunaDB.Client.csproj]: It's 2018 ;)

File: FaunaDB.Client/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -1,39 +1,5 @@
-﻿using System.Reflection;
-using System.Runtime.CompilerServices;
-
-// Information about this assembly is defined by the following attributes.
-// Change them to the values specific to your project.
-
-[assembly: AssemblyTitle("FaunaDB.Client")]
```
Comment (ashfire908) [inline FaunaDB.Client/Properties/AssemblyInfo.cs]: I take it all this is replaced by just the project files now?

File: FaunaDB.Client/Query/Expr.cs [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ﻿using Newtonsoft.Json;
 using System;
+using System.Reflection;
```
Comment (ashfire908) [inline FaunaDB.Client/Query/Expr.cs]: *blinks* was this an existing missing dep?

File: FaunaDB.Client/Types/Decoder.cs [hunk 4]
```diff
@@ -125,7 +125,10 @@ static object DecodeIntern(Value value, Type dstType)
                 case TypeCode.Decimal:
                     return Convert.ToDecimal(ToDouble<decimal>(value));
 
+#if !(NETSTANDARD1_5)
```
Comment (ashfire908) [inline FaunaDB.Client/Types/Decoder.cs]: Ew, but ok.

File: FaunaDB.Client/Types/Json.cs [hunk 5]
```diff
@@ -1,5 +1,6 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Reflection;
```
Comment (ashfire908) [inline FaunaDB.Client/Types/Json.cs]: Same as the other new dep in `Expr.cs`

File: FaunaDB.Client/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (marrony) [reply FaunaDB.Client/Properties/AssemblyInfo.cs]: yes

File: FaunaDB.Client/Query/Expr.cs (hunk 3 above)
Comment (marrony) [reply FaunaDB.Client/Query/Expr.cs]: no, this is necessary as it brings some extension methods for old frameworks

**Pull request merged**